config/migrations/20211208114312-subcase-name-lowercase.sparql

```diff
@@ -0,0 +1,56 @@
+PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
+PREFIX dossier: <https://data.vlaanderen.be/ns/dossier#>
+PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+
+DELETE {
+    GRAPH <http://mu.semte.ch/graphs/public> {
+        ?snelkoppeling skos:prefLabel ?subcaseNameOld .
+    }
+}
+INSERT {
+    GRAPH <http://mu.semte.ch/graphs/public> {
+        ?snelkoppeling skos:prefLabel ?subcaseNameNew .
+    }
+}
+WHERE {
+    GRAPH <http://mu.semte.ch/graphs/public> {
+        ?snelkoppeling a ext:SnelKoppeling .
+        ?snelkoppeling skos:prefLabel ?subcaseNameOld .
+        ?snelkoppeling ext:type "subcase-name" .
+
+        BIND (
+            CONCAT(
+                LCASE(SUBSTR(?subcaseNameOld, 1, 1)),
+                SUBSTR(?subcaseNameOld, 2)
+            ) AS ?subcaseNameNew)
+    }
+};
+
+DELETE {
+    GRAPH <http://mu.semte.ch/graphs/organizations/kanselarij> {
+        ?subcase ext:procedurestapNaam ?subcaseNameOld .
+    }
+}
+INSERT {
+    GRAPH <http://mu.semte.ch/graphs/organizations/kanselarij> {
+        ?subcase ext:procedurestapNaam ?subcaseNameNew .
+    }
+}
+WHERE {
+    GRAPH <http://mu.semte.ch/graphs/organizations/kanselarij> {
+        ?subcase a dossier:Procedurestap .
+        ?subcase ext:procedurestapNaam ?subcaseNameOld .
+    }
+
+    GRAPH <http://mu.semte.ch/graphs/public> {
+        ?snelkoppeling a ext:SnelKoppeling .
+        ?snelkoppeling skos:prefLabel ?subcaseNameNew .
+        ?snelkoppeling ext:type "subcase-name" .
+    }
+
+    BIND (
+        CONCAT(
+            LCASE(SUBSTR(?subcaseNameOld, 1, 1)),
+            SUBSTR(?subcaseNameOld, 2)
+        ) AS ?subcaseNameNew)
+}
```

Comment:
Dit stuk van de query lijkt mij fout.
Je gaat gewoon alle data dat nu in `ext:procedurestapNaam` zit overschrijven met een nieuwe naam van een (random?) snelkoppeling. Er is hier geen link tussen de procedurestap en de snelkoppeling

Kijk maar naar de select query resultaten (een stukje)
1 procedurestap komt tientallen keren voor.
![image](https://user-images.githubusercontent.com/22245223/147353408-f8684971-c91e-40f5-95b0-7a4eae28ea50.png)

Reply:
Volgens mij hoort dat stuk public graph niet thuis in deze 2e query

Reply:
zonder public stukje (op ACC data eens de select kant gerund)
![image](https://user-images.githubusercontent.com/22245223/147354131-07c01c4f-6601-4a8e-998f-2656253be3b5.png)

Reply:
1. Ge hebt gelijk. Blijkbaar is het de positie van `BIND` die het verschil maakt. `BIND` voor het `GRAPH`-pattern werkt wel.
2. Dat GRAPH-pattern was om custom (misschien correct behoofdletterde) namen niet te overschrijven.

Reply:
Deze query zonder het hele public stuk gaat alle procedurestappen namen aanpassen zodat deze starten zonder hoofdletter. In de frontend geven we die dan een hoofdletter.
Wat er voor de rest in die naam staat maakt niet uit, we behouden daar de hoofletters als die er zijn.

Reply:
Ik heb ook een fitler toegevoegd want er zijn lege strings als waarde soms, en die gaan mogelijk falen met de substring helper.